### PR TITLE
check missing docs for reexported macros as well

### LINKED ISSUE #56334: `#[deny(missing_docs)]` ignores exported `macro_rules!`
If my crate exports a `macro_rules!` macro, the `deny(missing_docs)` attribute has no effect even though it probably should.

The following code should, at the very least, provide a warning
```rust
#[deny(missing_docs)]
#[macro_export]
macro_rules! foo {
    () => {}
}
```

Preventing this code from compiling would likely break some backwards compatability, but I don't see why it shouldn't at least give a warning.

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Fixes #56334.

### CODE REVIEW

File: src/test/rustdoc-ui/deny-missing-docs.rs [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+#![deny(missing_docs)] //~ ERROR
+
+#[deny(missing_docs)]
+#[macro_export]
+macro_rules! foo {
+    () => {}
+}
```
Comment (Centril) [inline src/test/rustdoc-ui/deny-missing-docs.rs]: You'll get a tidy error?

**Comment by GuillaumeGomez** [comment]
I don't understand how the failure can be linked to this PR...

**Comment by ollie27** [comment]
The failure is because it's now checking for the `#[macro_export]` attribute on items that aren't macros so the `unused_attribute` lint thinks the attribute is used.

However as can be seen from the added ui test this hasn't actually worked. It's only giving a `error: missing documentation for crate` but it should be giving an error about the macro.

**Comment by GuillaumeGomez** [comment]
Thanks for the explanation @ollie27! I'll fix this in the next days.

**Comment by GuillaumeGomez** [comment]
cc @petrochenkov @eddyb

**Comment by GuillaumeGomez** [comment]
Arf, I need to make a PR to `stdsimd` too...

**Comment by Centril** [comment]
Re-assigning:

r? @varkor

File: src/libcore/lib.rs [hunk 2]
```diff
@@ -129,6 +129,7 @@
 #![feature(reverse_bits)]
 #![feature(non_exhaustive)]
 #![feature(structural_match)]
+#![feature(avx512_target_feature)]
```
Comment (varkor) [inline src/libcore/lib.rs]: What's the relevance of this feature (and the stdsimd update)?

**Review by varkor — COMMENTED** [review]
r=me if the `avx512_target_feature` feature is intentional (or without it if unnecessary).

File: src/libcore/lib.rs (hunk 2 above)
Comment (GuillaumeGomez) [reply src/libcore/lib.rs]: Without it we get:

```
error[E0658]: the target feature `avx512f` is currently unstable (see issue #44839)e
  --> src/libcore/../stdsimd/coresimd/x86/avx512f.rs:13:18
   |
13 | #[target_feature(enable = "avx512f")]
   |                  ^^^^^^^^^^^^^^^^^^
   |
   = help: add #![feature(avx512_target_feature)] to the crate attributes to enable

error[E0658]: the target feature `avx512f` is currently unstable (see issue #44839)
  --> src/libcore/../stdsimd/coresimd/x86/avx512f.rs:30:18
   |
30 | #[target_feature(enable = "avx512f")]
   |                  ^^^^^^^^^^^^^^^^^^
   |
   = help: add #![feature(avx512_target_feature)] to the crate attributes to enable

error[E0658]: the target feature `avx512f` is currently unstable (see issue #44839)
  --> src/libcore/../stdsimd/coresimd/x86/avx512f.rs:43:18
   |
43 | #[target_feature(enable = "avx512f")]
   |                  ^^^^^^^^^^^^^^^^^^
   |
   = help: add #![feature(avx512_target_feature)] to the crate attributes to enable

error[E0658]: the target feature `avx512f` is currently unstable (see issue #44839)
  --> src/libcore/../stdsimd/coresimd/x86/avx512f.rs:55:18
   |
55 | #[target_feature(enable = "avx512f")]
   |                  ^^^^^^^^^^^^^^^^^^
   |
   = help: add #![feature(avx512_target_feature)] to the crate attributes to enable

error[E0658]: the target feature `avx512f` is currently unstable (see issue #44839)
  --> src/libcore/../stdsimd/coresimd/x86/avx512f.rs:65:18
   |
65 | #[target_feature(enable = "avx512f")]
   |                  ^^^^^^^^^^^^^^^^^^
   |
   = help: add #![feature(avx512_target_feature)] to the crate attributes to enable

error: aborting due to 5 previous errors
```

**Comment by GuillaumeGomez** [comment]
To avoid new stdsmid update:

@bors: r+ p=1

**Comment by Mark-Simulacrum** [comment]
Attempting to poke bors

@bors retry r- r=varkor

**Comment by GuillaumeGomez** [comment]
For some reason it's not approved anymore...

@bors: r=varkor

**Comment by GuillaumeGomez** [comment]
Let's give it another try...

@bors: r=varkor p=10

**Pull request merged**